### PR TITLE
create a ClusterClientManager

### PR DESCRIPTION
**Pull request by wwitzel3**
First step in breaking out the global state of Octant

### CODE REVIEW

File: internal/config/dash.go [hunk 1]
```diff
@@ -94,49 +97,59 @@ type Dash interface {
 	Validate() error
 
 	ModuleManager() module.ManagerInterface
+
+	// ClusterClientManager()
```
Comment (bryanl) [inline internal/config/dash.go]: If this is going to be implemented in a future change, create a todo?

File: internal/config/dash.go [hunk 2]
```diff
@@ -94,49 +97,59 @@ type Dash interface {
...
+		kubeConfigPath:       kubeConfigPath,
+		logger:               logger,
+		moduleManager:        moduleManager,
+		objectStore:          objectStore,
+		pluginManager:        pluginManager,
+		portForwarder:        portForwarder,
+		currentContextName:   currentContextName, // remove
+		restConfigOptions:    restConfigOptions,
+	}
+	clusterClient, err := l.clusterClientManager.Get(context.Background(), currentContextName)
```
Comment (bryanl) [inline internal/config/dash.go]: This context should come from somewhere else. Don't use a `Background` here.

File: internal/config/dash.go (hunk 1 above)
Comment (wwitzel3) [reply internal/config/dash.go]: It is, it is, just a left over comment. Line 22 is identical (and uncommented)

File: internal/config/dash.go (hunk 2 above)
Comment (wwitzel3) [reply internal/config/dash.go]: Made it a TODO context for now, will get a proper context plumbed down to it.

File: internal/config/dash.go [hunk 3]
```diff
@@ -94,49 +97,59 @@ type Dash interface {
...
+	clusterClient        cluster.ClientInterface // clusterClientManager
+	clusterClientManager client.ClusterClientManager
+	crdWatcher           CRDWatcher
+	logger               log.Logger
+	moduleManager        module.ManagerInterface
+	objectStore          store.Store
+	pluginManager        plugin.ManagerInterface
+	portForwarder        portforward.PortForwarder
+	kubeConfigPath       string
+	currentContextName   string // remove .. replace with CurrentContext() method ..
```
[the remark this answers is not in the document]
Comment (wwitzel3) [reply internal/config/dash.go]: I can do it in the this PR, but my plan was to do this in a separate PR. I can just continue the work as a new commit on this same PR.

File: internal/config/dash.go [hunk 4]
```diff
@@ -186,23 +209,18 @@ func (l *Live) PortForwarder() portforward.PortForwarder {
 
 // UseContext switches context name. This process should have synchronously.
 func (l *Live) UseContext(ctx context.Context, contextName string) error {
-	client, err := cluster.FromKubeConfig(ctx, l.kubeConfigPath, contextName, l.restConfigOptions)
-	if err != nil {
-		return err
-	}
-
-	l.ClusterClient().Close()
```
[the remark this answers is not in the document]
Comment (wwitzel3) [reply internal/config/dash.go]: Yeah, I can address this in this PR. `newCluster` I think will become `Open` and I'll have the `FromKubeConfig` function give back a clientConfig / restConfig .. the cluster client manager can then store those and `Open` the a client interface on demand. As for closing, we could add a timeout mechanism that closes clients that have gone unused for a period of time.

**Pull request closed without merging**